core/src/main/java/org/apache/brooklyn/core/mgmt/ha/HighAvailabilityManagerImpl.java

```diff
@@ -1072,6 +1081,26 @@ private ManagementPlaneSyncRecord loadManagementPlaneSyncRecordInternal(boolean
         throw new IllegalStateException(message, lastException);
     }
 
+    private ManagementPlaneSyncRecord updateManagementPlaneSyncRecordWithLocalKnowledge(ManagementPlaneSyncRecord result) {
```

Comment:
I would add javadoc saying that this method re-assigns the value of the supplied parameter. Otherwise, make it final if you prefer to return a new value from this method.